api/openapi.yaml

```diff
@@ -1685,34 +1661,81 @@ components:
               items:
                 type: string
               nullable: true
-            state:
-              description: The lifecycle state of this scan.
-              type: string
-              enum:
-                - Pending
-                - Discovered
-                - InProgress
-                - Aborted
-                - Failed
-                - Done
-            stateMessage:
-              description: Human-readable message indicating details about the last state transition.
-              type: string
-            stateReason:
-              description: Machine-readable, UpperCamelCase text indicating the reason for the condition's last transition.
-              type: string
-              enum:
-                - Aborted
-                - TimedOut
-                - OneOrMoreAssetFailedToScan
-                - DiscoveryFailed
-                - Unexpected
-                - NothingToScan
-                - Success
+            status:
+              $ref: '#/components/schemas/ScanStatus'
             summary:
               $ref: '#/components/schemas/ScanSummary'
         - $ref: '#/components/schemas/ScanTemplate'
 
+    ScanStatus:
+      type: object
+      required:
+        - state
+        - reason
+        - lastTransitionTime
+      properties:
+        state:
+          description: |
+            Describes the state of Scan.
+            
+            | State      | Description                                       |
+            | ---------- | ------------------------------------------------- |
+            | Pending    | Scan is waiting to be scheduled                   |
+            | Discovered | Assets have been found for the Scan               |
+            | InProgress | The AssetScans are running on the target machines |
+            | Aborted    | Scan has been aborted through the API             |
+            | Failed     | At least one AssetScan ended with an error        |
+            | Done       | Every AssetScan ended successfully                |
```

Comment:
Similar to Kubernetes, we could have a bit more descriptive states:
- **Pending** - The initial state when the scan is created, and is working to discover assets
- **Discovered**: The asset scans have been discovered and are waiting to be scheduled
- **InProgress**: The asset scans have been scheduled and the assets are being scanned
- **Aborted**: The scan has been manually aborted
- **Failed**: One or more assets could not be successfully scanned
- **Done**: All assets have been scanned successfully

Later on, additional more granular states can be added to indicate transitions between different states e.g. `Discovered` to `InProgress` through `Scheduling`, `AssetScanCreating`, etc.

Reply:
@chrisgacsal please add input here because the `reason` messages do not need to be so contextually strict to understand the `state` on the status

Reply:
@chrisgacsal feel free to add input here because the `reason` messages do not need to be so contextually strict to understand the `state` on the status. wdyt?